app/session.server.ts

```diff
@@ -36,6 +36,22 @@ export async function getUsername(request: Request): Promise<User['username'] |
   return username;
 }
 
+export async function getEffectiveName(request: Request): Promise<string> {
+  const cookie = request.headers.get('Cookie');
+  const effectiveName = await effectiveNameCookie.parse(cookie);
+  return effectiveName;
+}
+
+export async function setEffectiveName(username: string, name: string): Promise<string> {
+  const user = await getUserByUsername(username);
+
+  if (user?.group.includes('admin')) {
```

Comment:
Let's fix the `User` returned in https://github.com/DevelopingSpace/starchart/blob/main/app/models/user.server.ts#L7-L21 and re-use/modify the code in https://github.com/DevelopingSpace/starchart/blob/main/app/models/user.server.ts#L61-L77 to add properties to the `User` object that indicate if this is an admin.

```
User.isAdmin
```

This way we can do the `/mycustomdomain(-dev)?-admins/.test(group)` stuff to set it on the object.

Reply:
Is making use of isAdmin when setting effectiveName enough or should I do as you said and make a boolean property on a user for whether they are Admin?

Reply:
Whatever you do, we need to put it on the `User` so we can pass that info around.  But using `isAdmin()` like you are is good internally for your needs.